dvc/tree/s3.py

```diff
@@ -227,6 +228,20 @@ def walk_files(self, path_info, **kwargs):
 
             yield path_info.replace(path=fname)
 
+    def ls(self, path_info, recursive=False, detail=False):
+        assert recursive
+
+        with self._get_bucket(path_info.bucket) as bucket:
+            for obj_summary in bucket.objects.filter(Prefix=path_info.path):
+                file_info = path_info.replace(path=obj_summary.key)
+                if detail:
+                    yield file_info, {
+                        "size": obj_summary.size,
+                        "etag": obj_summary.e_tag.strip('"'),
+                    }
```

Comment:
This should return just a dict with a string as a name. Creating lots of `file_info`s will be really costly.

Reply:
> Also doesn't comply with fsspec

I am aware of this, though I thought it would comply with the general sense of our trees (exchanging file_info's everywhere [walk_files, upload, download, etc.]).

Reply:
I see, you are right. But let's try to comply with fsspec if we can. This looks like a good place to do it, as you'll need to call `.replace` only once instead of in every `ls` implementation.